src/inference/dev_api/openvino/runtime/iplugin.hpp

```diff
@@ -185,6 +185,33 @@ class OPENVINO_RUNTIME_API IPlugin : public std::enable_shared_from_this<IPlugin
                                                              const ov::SoPtr<ov::IRemoteContext>& context,
                                                              const ov::AnyMap& properties) const = 0;
 
+    /**
+     * @brief Creates an compiled model from an previously exported model using plugin implementation
+     *        and removes OpenVINO Runtime magic and plugin name
+     * @param model Reference to model output stream
+     * @param weights_buffer AlignedBuffer with cached model
+     * @param properties A ov::AnyMap of properties
+     * @return An Compiled model
+     */
+    virtual std::shared_ptr<ov::ICompiledModel> import_model(std::istream& model,
+                                                             std::shared_ptr<ov::AlignedBuffer> model_buffer,
+                                                             const ov::AnyMap& properties) const;
```

Comment:
Why not to pass this `model_buffer` via properties ? It will require less changes

In future, we are moving to weightless cache for all plugins, so `model_buffer` will be less important.